pkg/apis/core/v1alpha1/g8s_control_plane_types.go

```diff
@@ -0,0 +1,128 @@
+package v1alpha1
+
+import (
+	"fmt"
+
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// NewG8sControlPlaneCRD returns a new custom resource definition for
+// G8sControlPlane. This might look something like the following.
+//
+//	apiVersion: apiextensions.k8s.io/v1beta1
+//	kind: CustomResourceDefinition
+//	metadata:
+//	  name: g8sControlPlanes.core.giantswarm.io
+//	spec:
+//	  group: core.giantswarm.io
```

Comment:
We should maybe make this part of the infrastructure type. /cc @tuommaki WDYT?

Reply:
You mean the group should be infrastructure? Yes.
```suggestion
//	  group: infrastructure.giantswarm.io
```